src/main/java/bio/terra/cli/service/utils/HttpUtils.java

```diff
@@ -11,9 +11,19 @@
 import java.net.URLEncoder;
 import java.nio.charset.Charset;
 import java.util.Map;
+import java.util.function.Predicate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /** Utility methods for making raw HTTP requests (e.g. in place of using a client library). */
 public class HttpUtils {
+  private static final Logger logger = LoggerFactory.getLogger(HttpUtils.class);
+
+  // default value for the maximum number of times to retry HTTP requests
+  private static final int DEFAULT_MAXIMUM_RETRIES = 10;
+
+  // default value for the number of milliseconds to sleep between retries
+  private static final int DEFAULT_MILLISECONDS_SLEEP_FOR_RETRY = 1000;
```

Comment:
nit: Prefer using java.time.Duration for durations e.g. `private static final Duration DEFAULT_SLEEP_FOR_RETRY = Duration.ofSeconds(1);`. I realize that we'll have to convert it `toMillis()` for Thread sleep, but it's still much less error prone to work in terms of Durations until then. (yes, this is a tiny use case, but good java time habits/the number of bugs because of primitive time durations.)

Reply:
ok